CoreFoundation/CMakeLists.txt

```diff
@@ -0,0 +1,348 @@
+                PlugIn.subproj/CFPlugIn.c
+                PlugIn.subproj/CFPlugIn_Factory.c
+                PlugIn.subproj/CFPlugIn_Instance.c
+                PlugIn.subproj/CFPlugIn_PlugIn.c
+                # Preferences
+                Preferences.subproj/CFApplicationPreferences.c
+                Preferences.subproj/CFPreferences.c
+                Preferences.subproj/CFXMLPreferencesDomain.c
+                # RunLoop
+                RunLoop.subproj/CFRunLoop.c
+                RunLoop.subproj/CFSocket.c
+                # Stream
+                Stream.subproj/CFConcreteStreams.c
+                Stream.subproj/CFSocketStream.c
+                Stream.subproj/CFStream.c
+                # String
+                String.subproj/CFAttributedString.c
+                String.subproj/CFBurstTrie.c
+                String.subproj/CFCharacterSet.c
+                String.subproj/CFCharacterSetData.S
+                String.subproj/CFRegularExpression.c
+                String.subproj/CFRunArray.c
+                String.subproj/CFString.c
+                String.subproj/CFStringEncodings.c
+                String.subproj/CFStringScanner.c
+                String.subproj/CFStringTransform.c
+                String.subproj/CFStringUtilities.c
+                String.subproj/CFUniCharPropertyDatabase.S
+                String.subproj/CFUnicodeData.S
+                # StringEncodings
+                StringEncodings.subproj/CFBuiltinConverters.c
+                StringEncodings.subproj/CFICUConverters.c
+                StringEncodings.subproj/CFPlatformConverters.c
+                StringEncodings.subproj/CFStringEncodingConverter.c
+                StringEncodings.subproj/CFStringEncodingDatabase.c
+                StringEncodings.subproj/CFUniChar.c
+                StringEncodings.subproj/CFUnicodeDecomposition.c
+                StringEncodings.subproj/CFUnicodePrecomposition.c
+                # URL
+                URL.subproj/CFURL.c
+                URL.subproj/CFURLAccess.c
+                URL.subproj/CFURLComponents.c
+                URL.subproj/CFURLComponents_URIParser.c)
+if(CMAKE_SYSTEM_NAME STREQUAL Linux)
+  target_compile_definitions(CoreFoundation
+                             PRIVATE
+                               -DDEPLOYMENT_TARGET_LINUX
+                               -D_GNU_SOURCE)
+elseif(CMAKE_SYSTEM_NAME STREQUAL FreeBSD)
+  target_compile_definitions(CoreFoundation
+                             PRIVATE
+                               -DDEPLOYMENT_TARGET_FREEBSD)
+elseif(CMAKE_SYSTEM_NAME STREQUAL Darwin)
+  target_compile_definitions(CoreFoundation
+                             PRIVATE
+                               -DDEPLOYMENT_TARGET_MACOSX)
+endif()
+target_compile_definitions(CoreFoundation
+                           PRIVATE
+                             -DU_SHOW_DRAFT_API
+                             -DCF_BUILDING_CF
+                             -DDEPLOYMENT_RUNTIME_C
```

Comment:
only one deployment runtime should be specified; if you are building for swift you need the RUNTIME_SWIFT version and if you are building for stand-alone you need the RUNTIME_C version.

Reply:
it could also be a bug btw ;)